### PR TITLE
Added unit tests for profile updates

### PR DESCRIPTION
**Pull request by slintes**
Added unit tests for profile updates.

### CODE REVIEW

**Comment by slintes** [comment]
wait for #73 

/hold

**Comment by slintes** [comment]
rebased and improved test for updating hugepages with numa node

**Comment by slintes** [comment]
modified after recent API changes, ready for review :)

/hold cancel

File: pkg/controller/performanceprofile/performanceprofile_controller_test.go [hunk 1]
```diff
@@ -167,24 +160,14 @@ var _ = Describe("Controller", func() {
...
 			Expect(err).ToNot(HaveOccurred())
 		})
 
-		It("should do nothing on third reconcile loop", func() {
+		It("should create event on third reconcile loop", func() {
 			r := newFakeReconciler(profile)
 
-			result, err := r.Reconcile(request)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(result).To(Equal(reconcile.Result{RequeueAfter: 10 * time.Second}))
```
Comment (davidvossel) [inline pkg/controller/performanceprofile/performanceprofile_controller_test.go]: right here we're checking that the first reconcile loop results in a requeue, and that the others do not. How is that handled now with the new "reconcileTimes()" function?

Comment (slintes) [reply pkg/controller/performanceprofile/performanceprofile_controller_test.go]: the requeue should always happen on the 1st reconcile loop, and that one is tested in line 112

File: pkg/controller/performanceprofile/performanceprofile_controller_test.go [hunk 2]
```diff
@@ -101,16 +102,14 @@ var _ = Describe("Controller", func() {
...
 			Expect(errors.IsNotFound(err)).To(Equal(true))
 		})
 
 		It("should create all resources except KubeletConfig on first reconcile loop", func() {
 			r := newFakeReconciler(profile)
 
-			result, err := r.Reconcile(request)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(result).To(Equal(reconcile.Result{RequeueAfter: 10 * time.Second}))
+			Expect(reconcileTimes(r, request, 1)).To(Equal(reconcile.Result{RequeueAfter: 10 * time.Second}))
```
Comment (slintes) [inline pkg/controller/performanceprofile/performanceprofile_controller_test.go]: @davidvossel see here

**Pull request merged**